clients/rest-client-generator/src/test/java/uk/gov/justice/services/clients/rest/generator/RestClientGenerator_MethodBodyTest.java

```diff
@@ -0,0 +1,193 @@
+import uk.gov.justice.services.adapters.test.utils.builder.ActionBuilder;
+import uk.gov.justice.services.adapters.test.utils.compiler.JavaCompilerUtil;
+import uk.gov.justice.services.clients.core.EndpointDefinition;
+import uk.gov.justice.services.clients.core.RestClientHelper;
+import uk.gov.justice.services.clients.core.RestClientProcessor;
+import uk.gov.justice.services.messaging.DefaultJsonEnvelope;
+import uk.gov.justice.services.messaging.JsonEnvelope;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.hamcrest.beans.HasPropertyWithValue.hasProperty;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsCollectionContaining.hasItems;
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.same;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.raml.model.ActionType.GET;
+import static uk.gov.justice.services.adapters.test.utils.builder.ActionBuilder.defaultGetAction;
+import static uk.gov.justice.services.adapters.test.utils.builder.RamlBuilder.restRamlWithDefaults;
+import static uk.gov.justice.services.adapters.test.utils.builder.RamlBuilder.restRamlWithQueryApiDefaults;
+import static uk.gov.justice.services.adapters.test.utils.builder.ResourceBuilder.defaultGetResource;
+import static uk.gov.justice.services.adapters.test.utils.builder.ResourceBuilder.resource;
+import static uk.gov.justice.services.adapters.test.utils.config.GeneratorConfigUtil.configurationWithBasePackage;
+import static uk.gov.justice.services.adapters.test.utils.reflection.ReflectionUtil.firstMethodOf;
+import static uk.gov.justice.services.adapters.test.utils.reflection.ReflectionUtil.setField;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RestClientGenerator_MethodBodyTest {
+    private static final String BASE_PACKAGE = "org.raml.test";
+    private static final JsonEnvelope NOT_USED_ENVELOPE = DefaultJsonEnvelope.envelopeFrom(null, null);
+    private static final Map<String, String> NOT_USED_GENERATOR_PROPERTIES = ImmutableMap.of("serviceComponent", "QUERY_CONTROLLER");
+
+    private RestClientGenerator restClientGenerator;
+
+    private JavaCompilerUtil compiler;
+
+    @Mock
+    RestClientProcessor restClientProcessor;
+
+    @Mock
+    RestClientHelper restClientHelper;
+
+    @Rule
+    public TemporaryFolder outputFolder = new TemporaryFolder();
+
+    @Before
+    public void before() {
+
+        restClientGenerator = new RestClientGenerator();
+        compiler = new JavaCompilerUtil(outputFolder.getRoot(), outputFolder.getRoot());
+    }
+
+    @Test
+    public void shouldCallRestClientWithEndpointDefinitionContainingBaseUri() throws Exception {
+
+        restClientGenerator.run(
```

Comment:
Possibly, but as above the tests tend to be subtly different so hard to achieve without obscuring what is being tested